OpenSim/Simulation/Model/Model.cpp

```diff
@@ -661,9 +661,15 @@ void Model::createMultibodyTree()
 
     auto bodies = getComponentList<Body>();
     for (auto& body : bodies) {
-        _multibodyTree.addBody( body.getAbsolutePathName(),
-                                body.getMass(), false,
-                                const_cast<Body*>(&body) );
+        try {
+            _multibodyTree.addBody( body.getAbsolutePathName(),
+                                    body.getMass(), false,
+                                    const_cast<Body*>(&body) );
+        } catch (const std::runtime_error& e) {
+            cout << "Duplicate Body names are not permitted. Please update "
+                 << "your model's .osim file accordingly." << std::endl;
```

Comment:
> Could one get this message when building a model programmatically (rather than loading an .osim file)? Consider adding to the message something like "

I thought that would be caught elsewhere. I thought this issue had been investigated so just added a slightly modified version of the text @jenhicks requested in #1768. I will investigate thoroughly and revise as necessary.

> Is it that names must not be duplicated, or that absolute path names may not be duplicated? The message makes sense for pre-4.0 models, but is not really correct for 4.0 models.

The string being passed to Simbody's MultibodyGraphMaker is actually the absolute path name, so I agree that the message requested in #1768 is not complete. I will revise.

> Is there a reason you chose to send a message to cout rather than throw a new exception with a more specific message?

The request in #1768 was to "add to the message…". OpenSim doesn't have control over the errors that Simbody throws. In theory, Simbody could throw a `runtime_error` from `addBody()` (or one of the methods it calls) with some other message, so I think it's safest to throw the original error after adding an OpenSim-specific message. Actually, the message from OpenSim should say "Perhaps there are duplicate names…" or something like that just in case Simbody threw a `runtime_error` for some other reason. I will revise.